backend/internal/storage/job_offers_db.go

```diff
@@ -89,20 +98,33 @@ func (db DB) PostJobOffer(query v1beta.OfferPostQuery) (*v1beta.JobOffer, error)
 		}
 
 		offer.TitleID = jobTitle.ID
-		res := tx.Table("job_offers").Create(&offer)
+		res := tx.Table("job_offers").Create(offer)
 		if res.Error != nil {
 			log.Error(res.Error)
 			return res.Error
 		}
 
+		if uploadImageFunc != nil {
+			offer.CompanyImageLocation, err = uploadImageFunc(offer.ID)
+			if err != nil {
+				log.Error(err)
+				return err
+			}
+
+			err = tx.Table("job_offers").Where("id = ?", offer.ID).Update("company_image_location", offer.CompanyImageLocation).Error
+			if err != nil {
+				log.Error(err)
+				return err
+			}
+		}
+
```

Comment:
As discussed on telegram, we would rather have an image table than add a link to the image location in the `joboffer` table, that will help us get rid of the callback and the circular dependency